invenio_rdm_records/services/components/pids.py

```diff
@@ -212,6 +212,11 @@ def delete_record(self, identity, data=None, record=None, uow=None):
             self.service.pids.parent_pid_manager.discard_all(
                 parent_pids, soft_delete=True
             )
+        else:
+            # We're sending a task in case there is a race condition with two
+            # versions being deleted at the same time to ensure that we have
+            # consistent database state
+            self.uow.register(TaskOp(cleanup_parent_pids, record["id"]))
```

Comment:
It looks like that the `if` above is copy/pasted in the celery task.
Does it make sense to completely remove the `if` condition here and simply call:
`self.uow.register(TaskOp(cleanup_parent_pids, record["id"]))`?

This is a bit confusing.